gradio/interface.py

```diff
@@ -476,7 +477,80 @@ def __repr__(self):
         return repr
 
     def get_config_file(self):
-        return utils.get_config_file(self)
+        components = []
+        layout = {"id": 0, "children": []}
+        dependencies = []
+
+        def add_component(parent, component):
+            id = len(components) + 1
+            components.append(
+                {
+                    "id": len(components) + 1,
+                    "type": component.__class__.__name__.lower(),
+                    "props": component.get_template_context(),
+                }
+            )
+            layout_context = {"id": id}
+            if isinstance(component, BlockContext):
+                layout_context["children"] = []
+            parent["children"].append(layout_context)
+            return layout_context
+
+        if self.title:
+            add_component(layout, Markdown("<h1>" + self.title + "</h1>"))
+        if self.description:
+            add_component(layout, Markdown(self.description))
+        panel_row = add_component(layout, Row())
+        input_panel = add_component(panel_row, Column())
+        input_ids = []
+        for component in self.input_components:
+            input_id = add_component(input_panel, component)
+            input_ids.append(input_id["id"])
+        input_panel_btns = add_component(input_panel, Row())
+        submit_btn = add_component(input_panel_btns, Button("Submit"))
+        clear_btn = add_component(input_panel_btns, Button("Clear"))
+
+        output_panel = add_component(panel_row, Column())
+        output_ids = []
+        for component in self.output_components:
+            output_id = add_component(output_panel, component)
+            output_ids.append(output_id["id"])
+        output_panel_btns = add_component(output_panel, Row())
+        flag_btn = add_component(output_panel_btns, Button("Flag"))
+        dependencies.append(
+            {
+                "id": 0,
+                "trigger": "click",
+                "targets": [submit_btn["id"]],
+                "inputs": input_ids,
+                "outputs": output_ids,
+            }
+        )
+        dependencies.append(
+            {
+                "id": 1,
+                "trigger": "click",
+                "targets": [clear_btn["id"]],
+                "inputs": [],
+                "outputs": input_ids + output_ids,
+            }
+        )
+        dependencies.append(
+            {
+                "id": 2,
+                "trigger": "click",
+                "targets": [flag_btn["id"]],
+                "inputs": input_ids + output_ids,
+                "outputs": [],
```

Comment:
how will flag button work? completely in frontend?

Reply:
Still needs to be implemented in backend, but now handled from the /api/predict call, need to connect flagging handler to it.